protos/core.proto

```diff
@@ -12,6 +12,7 @@ message ProjectConfig {
   string assertion_schema = 5;
 
   string schema_suffix = 7;
+  int32 idempotent_action_retries = 8;
```

Comment:
nit: let's put it above `schema_suffix` which folks aren't really supposed to set in `dataform.json`.